app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Api::V1::UsersController < ApplicationController
+  def index
+    return nil if User.all.empty?
+    render json: UserSerializer.new(User.all)
+  end
+
+  def show
+    return nil if !User.exist?(params[:id])
+    render json: UserSerializer.new(User.find(params[:id]))
+  end
+
+  def create
+    user = User.new(user_params)
+    return nil if !user.save
+    render json: UserSerializer.new(user)
+  end
+
+  def update
+    return nil if !User.exist?(params[:id])
+    render json: UserSerializer.new(User.update(params[:id], user_params))
+  end
+
+  def destroy
+    return nil if !User.exist?(params[:id])
+    User.destroy(params[:id])
+  end
+
+  private
+
+  def user_params
+    params.permit(:email, :provider, :token, :refresh_token)
+  end
+end
```

Comment:
Really well done with this controller @hashmaster3k !! 

I didn't get to doing sad paths on rails_engine so I'm curious as to your choice for returning `nil` in order to get a 204 HTTP status is that convention?

Reply:
Thank you. I think it depends on the action. According to JSON specification guide, return a 404 if something isn't found when fetching data (i.e user id 9999 doesn't exist). However, I felt that returning a blank page might be slightly better than a bunch of errors. If you'd like me to change it I can.

Reply:
Nope I agree. I think Ian mentioned something to that effect during my last project review. I asked out of curiosity!